dangerzone/updater/cosign.py

```diff
@@ -0,0 +1,165 @@
+import subprocess
+from pathlib import Path
+from tempfile import NamedTemporaryFile
+
+from ..container_utils import subprocess_run
+from ..util import get_resource_path
+from . import errors, log
+
+"""
+This module exposes functions to interact with the embedded cosign binary.
+"""
+
+_COSIGN_BINARY = str(get_resource_path("vendor/cosign").absolute())
+
+
+def verify_local_image(oci_image_folder: Path, pubkey: Path) -> None:
+    """Verify the given path against the given public key"""
+    cmd = [
+        _COSIGN_BINARY,
+        "verify",
+        "--key",
+        str(pubkey),
+        "--offline",
+        "--local-image",
+        str(oci_image_folder),
+    ]
+    log.debug(" ".join(cmd))
+    result = subprocess_run(cmd, capture_output=True)
+    if result.returncode != 0:
+        log.info(f"Failed to verify signature: {result.stderr.decode()}")
+        raise errors.SignatureVerificationError
+
+
+def verify_blob(pubkey: Path, bundle: str, payload: str) -> None:
+    cmd = [
+        _COSIGN_BINARY,
+        "verify-blob",
+        "--key",
+        str(pubkey.absolute()),
+        "--bundle",
+        bundle,
+        payload,
+    ]
+    log.debug(" ".join(cmd))
+    result = subprocess_run(cmd, capture_output=True)
+    # If the process return code is not 0, or doesn't contain the expected
+    # string, we raise an error.
+    if result.returncode != 0 or result.stderr != b"Verified OK\n":
```

Comment:
@apyrgio Just as an FIY:

After bumping the version of cosign, I've started to get some errors related to this, because they're using the stderr to add some info there.

```
        if result.returncode != 0 or result.stderr != b"Verified OK\n":
            log.debug("Failed to verify signature", result)
>           raise errors.SignatureVerificationError("Failed to verify signature", result)
E           dangerzone.updater.errors.SignatureVerificationError: ('Failed to verify signature', CompletedProcess(args=['D:\\a\\dangerzone\\dangerzone\\share\\vendor\\cosign', 'verify-blob', '--key', 'D:\\a\\dangerzone\\dangerzone\\tests\\assets\\test.pub.key', '--bundle', 'C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\tmpeir_046y', 'C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\tmpm917e9ep'], returncode=0, stdout=b'', stderr=b'setting TUF refresh period to 24h0m0s\nVerified OK\n'))
```

So I changed this statement to:

```suggestion
    if result.returncode != 0 or b"Verified OK\n" not in result.stderr:
```

Reply:
... and finally went to trust the process return number instead!